addon/ng2/blueprints/component/files/__path__/__name__.component.ts

```diff
@@ -0,0 +1,16 @@
+import {Component} from 'angular2/core';
+
+@Component({
+  moduleId: __moduleName,
+  selector: '<%= dasherizedModuleName %>',
+  templateUrl: '<%= dasherizedModuleName %>.component.html',
+  styleUrls: ['<%= dasherizedModuleName %>.component.<%= styleExt %>']
+})
+export class <%= classifiedModuleName %>Component {
```

Comment:
I think it comes down to the use case, what fraction of generated components will end up needing/using this? Unfortunately at this point we don't have enough of a sample size to make a decision based upon facts/research only based upon gut feeling and the risk of devs being aware of the lifecycle hooks for components. I've seen enough examples of initialization code inside of the ctor, which is what lead me to add this in the first place.

Personally I think it should be there (with the interface reference also added), but I am also OK with it not being there at all.